Orange/widgets/visualize/owsieve.py

```diff
@@ -403,15 +409,22 @@ def _oper(attr, txt):
         view = self.canvasView
 
         chi = ChiSqStats(self.discrete_data, disc_x, disc_y)
-        n = chi.n
         max_ylabel_w = max((width(val) for val in disc_y.values), default=0)
         max_ylabel_w = min(max_ylabel_w, 200)
         x_off = width(attr_x.name) + max_ylabel_w
         y_off = 15
-        square_size = min(view.width() - x_off - 35, view.height() - y_off - 50)
+        square_size = min(view.width() - x_off - 35, view.height() - y_off - 80)
         square_size = max(square_size, 10)
         self.canvasView.setSceneRect(0, 0, view.width(), view.height())
-
+        if not disc_x.values or not disc_y.values:
+            text_ = "Features {} and {} have no values".format(disc_x, disc_y) \
+                if not disc_x.values and not disc_y.values and \
+                   disc_x != disc_y else "Feature {} has no values".format(
+                disc_x if not disc_x.values else disc_y)
```

Comment:
Could you use a normal `if` statement, not expression? I guess it would be so much easier to read. Refuse if you disagree.